notebooks/ssd-object-detection-demo.ipynb

```diff
@@ -0,0 +1,621 @@
+      "Output label prediction size: torch.Size([128, 81, 8732])\n",
+      "Average batch time: 366.42 ms\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Obtain the average time taken by a batch of input\n",
+    "benchmark(traced_model, input_shape=(128, 3, 300, 300), nruns=5000)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "---\n",
+    "<a id=\"4\"></a>\n",
+    "## 4. Compiling with TRTorch\n",
+    "TorchScript modules behave just like normal PyTorch modules and are intercompatable. From TorchScript we can now compile a TensorRT based module. This module will still be implemented in TorchScript but all the computation will be done in TensorRT."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import trtorch\n",
+    "\n",
+    "trt_model = trtorch.compile(traced_model, {\n",
+    "    \"input_shapes\": [(3, 3, 300, 300)],\n",
+    "    \"op_precision\": torch.half,\n",
+    "    \"workspace_size\": 1 << 20\n",
+    "})"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "---\n",
+    "<a id=\"5\"></a>\n",
+    "## 5. Running Inference"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can now run inference on the TRTorch compiled model. This is demonstrated below using sample images from the COCO 2017 Validation set."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "Using cache found in /root/.cache/torch/hub/NVIDIA_DeepLearningExamples_torchhub\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Sample images from the COCO validation set\n",
+    "uris = [\n",
+    "    'http://images.cocodataset.org/val2017/000000397133.jpg',\n",
+    "    'http://images.cocodataset.org/val2017/000000037777.jpg',\n",
+    "    'http://images.cocodataset.org/val2017/000000252219.jpg'\n",
+    "]\n",
+    "\n",
+    "# For convenient and comprehensive formatting of input and output of the model, load a set of utility methods.\n",
+    "utils = torch.hub.load('NVIDIA/DeepLearningExamples:torchhub', 'nvidia_ssd_processing_utils')\n",
+    "\n",
+    "# Format images to comply with the network input\n",
+    "inputs = [utils.prepare_input(uri) for uri in uris]\n",
+    "tensor = utils.prepare_tensor(inputs, False)\n",
+    "\n",
+    "# The model was trained on COCO dataset, which we need to access in order to\n",
+    "# translate class IDs into object names. \n",
+    "classes_to_labels = utils.get_coco_object_dictionary()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 12,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Run object detection\n",
+    "detections_batch = trt_model(tensor.to(torch.half))"
```

Comment:
Can we have a line here about how TRTorch module usage is exactly the same as normal PyTorch inference?

Reply:
Added. Thanks for the input.

Reply:
Added in latest commit.